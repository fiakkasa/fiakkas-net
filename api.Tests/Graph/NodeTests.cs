using HotChocolate.Execution;

namespace api.Tests.Graph;

public class NodeTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task Nodes_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              nodes(
                ids: [
                  #Achievement
                  "QWNoaWV2ZW1lbnQKZ2Q0NjA1YjBjNThiYzQ5YWNiY2ZkMTBhMjRhMjAzYWRk"
                  #ContactItem
                  "Q29udGFjdEl0ZW0KZ2ViZjIyNGE4N2ZmMzQ3Yjk4ODJiZGQ0MWVjN2Y1YTA1"
                  #Customer
                  "Q3VzdG9tZXIKZzE4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5"
                  #EducationItem
                  "RWR1Y2F0aW9uSXRlbQpnMzg4OThjNjIxNjFlNDBmMjhhOWYzOWJmMWZmNDYyMjQ="
                  #InformationTechnologyCategory
                  "SW5mb3JtYXRpb25UZWNobm9sb2d5Q2F0ZWdvcnkKZzZjYzQzZTlhMzEyYjQ5MjNiODkwZTk2NmI4MTY4ZWVl"
                  #Language
                  "TGFuZ3VhZ2UKZzAyYTNiZTliM2YwNDRiNGE4OTQ1ZTg0ZmVmNTM3YjU4"
                  #OtherCategory
                  "T3RoZXJDYXRlZ29yeQpnOWZkOTFmOGEyYTQ0NDUyMGE1YjE4YWExYzNjMjkxMzM="
                  #PortfolioCategory
                  "UG9ydGZvbGlvQ2F0ZWdvcnkKZzM4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5"
                  #PortfolioItem
                  "UG9ydGZvbGlvSXRlbQpnMjhlNDgzZTQ2OTYxNGIyNTg4YTlkMWQwYTUxNjExMDk="
                  #ResumeCategory
                  "UmVzdW1lQ2F0ZWdvcnkKZ2ViOWQ2MjU4OTljNDQ2YmRiZDQ0MjNkMzViMTk5NjVk"
                  #SoftwareDevelopmentCategory
                  "U29mdHdhcmVEZXZlbG9wbWVudENhdGVnb3J5CmdjYTgzMmJmOWI3Y2I0YzMxYmY4ZDAwZjg3YTI3NmZlMw=="
                  #TextItem
                  "VGV4dEl0ZW0KZzQ4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5"
                  #UnknownCategory
                  "VW5rbm93bkNhdGVnb3J5CmdjOWY1ODc5ZDQwMTg0OWEwOWI3MWI0NzlkZDVkZTdmZg=="
                  #UnknownTechnologyCategory
                  "VW5rbm93blRlY2hub2xvZ3lDYXRlZ29yeQpnZTBlOWNhNWQ1MjYzNDRmNzhiMjNiM2NlMjAzMjMxZDQ="
                ]
              ) {
                __typename
                id
                ... on Achievement {
                  content
                  createdAt
                  id
                  internalId
                  updatedAt
                  version
                  years
                  yearsSummary
                }
                ... on ContactItem {
                  createdAt
                  description
                  href
                  icon
                  id
                  internalId
                  key
                  title
                  updatedAt
                  version
                }
                ... on Customer {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on EducationItem {
                  categoryId
                  createdAt
                  description
                  href
                  id
                  internalId
                  location
                  subjects
                  title
                  updatedAt
                  version
                }
                ... on InformationTechnologyCategory {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on Language {
                  createdAt
                  id
                  internalId
                  proficiency
                  title
                  updatedAt
                  version
                }
                ... on OtherCategory {
                  createdAt
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on PortfolioCategory {
                  createdAt
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on PortfolioItem {
                  categoryId
                  createdAt
                  customerId
                  href
                  id
                  internalId
                  technologiesSummary
                  technologyIds
                  title
                  updatedAt
                  version
                  year
                }
                ... on ResumeCategory {
                  associatedCategoryTypes
                  createdAt
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on SoftwareDevelopmentCategory {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on TextItem {
                  content
                  createdAt
                  id
                  internalId
                  key
                  title
                  updatedAt
                  version
                }
                ... on UnknownCategory {
                  createdAt
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
                ... on UnknownTechnologyCategory {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                }
              }
            }
            """);

        var fn = result.ExpectOperationResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
