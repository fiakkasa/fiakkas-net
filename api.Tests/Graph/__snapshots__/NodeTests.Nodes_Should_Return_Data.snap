{
  "data": {
    "nodes": [
      {
        "__typename": "Achievement",
        "id": "QWNoaWV2ZW1lbnQKZ2Q0NjA1YjBjNThiYzQ5YWNiY2ZkMTBhMjRhMjAzYWRk",
        "content": "Content",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "d4605b0c-58bc-49ac-bcfd-10a24a203add",
        "updatedAt": null,
        "version": 1,
        "years": [
          2024
        ],
        "yearsSummary": "2024"
      },
      {
        "__typename": "ContactItem",
        "id": "Q29udGFjdEl0ZW0KZ2ViZjIyNGE4N2ZmMzQ3Yjk4ODJiZGQ0MWVjN2Y1YTA1",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "description": "Content",
        "href": "/test",
        "icon": "Icon",
        "internalId": "ebf224a8-7ff3-47b9-882b-dd41ec7f5a05",
        "key": "Key",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "Customer",
        "id": "Q3VzdG9tZXIKZzE4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "href": "/test",
        "internalId": "18e483e4-6961-4b25-88a9-d1d0a5161109",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "EducationItem",
        "id": "RWR1Y2F0aW9uSXRlbQpnMzg4OThjNjIxNjFlNDBmMjhhOWYzOWJmMWZmNDYyMjQ=",
        "categoryId": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "description": "Description",
        "href": "/test",
        "internalId": "38898c62-161e-40f2-8a9f-39bf1ff46224",
        "location": "Location",
        "subjects": [
          "Subject"
        ],
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "InformationTechnologyCategory",
        "id": "SW5mb3JtYXRpb25UZWNobm9sb2d5Q2F0ZWdvcnkKZzZjYzQzZTlhMzEyYjQ5MjNiODkwZTk2NmI4MTY4ZWVl",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "href": "/test",
        "internalId": "6cc43e9a-312b-4923-b890-e966b8168eee",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "Language",
        "id": "TGFuZ3VhZ2UKZzAyYTNiZTliM2YwNDRiNGE4OTQ1ZTg0ZmVmNTM3YjU4",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "02a3be9b-3f04-4b4a-8945-e84fef537b58",
        "proficiency": "NATIVE",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "OtherCategory",
        "id": "T3RoZXJDYXRlZ29yeQpnOWZkOTFmOGEyYTQ0NDUyMGE1YjE4YWExYzNjMjkxMzM=",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "9fd91f8a-2a44-4520-a5b1-8aa1c3c29133",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "PortfolioCategory",
        "id": "UG9ydGZvbGlvQ2F0ZWdvcnkKZzM4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "38e483e4-6961-4b25-88a9-d1d0a5161109",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "PortfolioItem",
        "id": "UG9ydGZvbGlvSXRlbQpnMjhlNDgzZTQ2OTYxNGIyNTg4YTlkMWQwYTUxNjExMDk=",
        "categoryId": "38e483e4-6961-4b25-88a9-d1d0a5161109",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "customerId": "18e483e4-6961-4b25-88a9-d1d0a5161109",
        "href": "/test",
        "internalId": "28e483e4-6961-4b25-88a9-d1d0a5161109",
        "technologiesSummary": "Title",
        "technologyIds": [
          "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3"
        ],
        "title": "Title",
        "updatedAt": null,
        "version": 1,
        "year": 2024
      },
      {
        "__typename": "ResumeCategory",
        "id": "UmVzdW1lQ2F0ZWdvcnkKZ2ViOWQ2MjU4OTljNDQ2YmRiZDQ0MjNkMzViMTk5NjVk",
        "associatedCategoryTypes": [
          "SOFTWARE_DEVELOPMENT"
        ],
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "eb9d6258-99c4-46bd-bd44-23d35b19965d",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "SoftwareDevelopmentCategory",
        "id": "U29mdHdhcmVEZXZlbG9wbWVudENhdGVnb3J5CmdjYTgzMmJmOWI3Y2I0YzMxYmY4ZDAwZjg3YTI3NmZlMw==",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "href": "/test",
        "internalId": "ca832bf9-b7cb-4c31-bf8d-00f87a276fe3",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "TextItem",
        "id": "VGV4dEl0ZW0KZzQ4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5",
        "content": "Content",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "48e483e4-6961-4b25-88a9-d1d0a5161109",
        "key": "Key",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      {
        "__typename": "UnknownCategory",
        "id": "VW5rbm93bkNhdGVnb3J5CmdjOWY1ODc5ZDQwMTg0OWEwOWI3MWI0NzlkZDVkZTdmZg==",
        "createdAt": "2024-01-01T00:00:00.000Z",
        "internalId": "c9f5879d-4018-49a0-9b71-b479dd5de7ff",
        "title": "Title",
        "updatedAt": null,
        "version": 1
      },
      null
    ]
  }
}
